src/snowflake/snowpark/mock/functions.py

```diff
@@ -393,3 +393,8 @@ def mock_to_timestamp(column: ColumnEmulator, fmt: Union[ColumnEmulator, str] =
         sf_type=ColumnType(TimestampType(), column.sf_type.nullable),
         dtype=object,
     )
+
+
+@patch("row_number")
+def mock_row_number(window: TableEmulator, row_idx: int):
```

Comment:
Another PR is for windowing function. I'm curious if that is required to have the `row_number` work.

Reply:
Yes, this is a stacked PR, I'm basing off this branch on [local/support-window-functions](https://github.com/snowflakedb/snowpark-python/tree/local/support-window-functions)